cmd/contour/serve.go

```diff
@@ -672,76 +662,69 @@ func (s *Server) setupGatewayAPI(contourConfiguration contour_api_v1alpha1.Conto
 
 	// Check if GatewayAPI is configured.
 	if contourConfiguration.Gateway != nil {
+		// Create and register the gatewayclass controller with the manager.
+		gatewayClassControllerName := contourConfiguration.Gateway.ControllerName
+		if _, err := controller.NewGatewayClassController(
+			mgr,
+			eventHandler,
+			sh.Writer(),
+			s.log.WithField("context", "gatewayclass-controller"),
+			gatewayClassControllerName,
+			isLeader,
+		); err != nil {
+			s.log.WithError(err).Fatal("failed to create gatewayclass-controller")
+		}
 
-		// Only inform on GatewayAPI if found in the cluster.
-		if s.clients.ResourcesExist(k8s.GatewayAPIResources()...) {
```

Comment:
I dropped the explicit resources-exist check since it seemed redundant now: startup will fail anyway if you try to enable Gateway API but the resources don't exist. Had the benefit of being able to entirely get rid of the RESTMapper.